### PR TITLE
Add voice message support

### PR DESCRIPTION
**Pull request by flazepe**
This commit adds the new properties for voice messages and functions to the `Attachment` struct. `AttachmentFlags` is also added as an extra.

### CODE REVIEW

File: src/structs/channels.rs [hunk 1]
```diff
@@ -543,6 +549,17 @@ bitflags! {
     const FAILED_TO_MENTION_SOME_ROLES_IN_THREAD = 1 << 8;
     /// This message will not trigger push and desktop notifications
     const SUPPRESS_NOTIFICATIONS = 1 << 12;
+    /// This message is a voice message
+    const IS_VOICE_MESSAGE = 1 << 12;
+  }
+}
+
+bitflags! {
+  /// Bitflags for Discord Message Flags
```
Comment (tonkku107) [inline src/structs/channels.rs:558]: These are attachment flags, not message flags :)

File: src/structs/channels.rs [hunk 2]
```diff
@@ -1337,9 +1357,24 @@ impl Attachment {
...
       width: None,
-      ephemeral: None
+      ephemeral: None,
+      duration_secs: None,
+      waveform: None,
+      flags: None
     }
   }
+
+  /// Sets the duration of the file in seconds (currently for voice messages)
```
Comment (tonkku107) [inline src/structs/channels.rs:1367]: copying the `(currently for voice messages)` text isn't necessary here

File: src/structs/channels.rs [hunk 3]
```diff
@@ -543,6 +549,17 @@ bitflags! {
     const FAILED_TO_MENTION_SOME_ROLES_IN_THREAD = 1 << 8;
     /// This message will not trigger push and desktop notifications
     const SUPPRESS_NOTIFICATIONS = 1 << 12;
+    /// This message is a voice message
+    const IS_VOICE_MESSAGE = 1 << 12;
```
Comment (tonkku107) [inline src/structs/channels.rs:553]: voice messages are `1 << 13`

**Review by tonkku107 — CHANGES_REQUESTED** [review]

File: src/structs/channels.rs (hunk 3 above)
Comment (flazepe) [reply src/structs/channels.rs:553]: Oops, I copy pasted the last line but only changed the name :p

File: src/structs/channels.rs [hunk 4]
```diff
@@ -1337,9 +1357,24 @@ impl Attachment {
...
-      ephemeral: None
+      ephemeral: None,
+      duration_secs: None,
+      waveform: None,
+      flags: None
     }
   }
+
+  /// Sets the duration of the file in seconds
+  pub fn set_duration_secs(mut self, duration_secs: f64) -> Self {
```
Comment (tonkku107) [inline src/structs/channels.rs:1368]: Uploading a file is done through the [`File`](https://github.com/tonkku107/slashook/blob/c58ff0b1e6d4fbbb0288605986ea57f18708e3ff/src/structs/utils.rs#L31) struct and an attachment is only created if necessary at the [rest multipart handler](https://github.com/tonkku107/slashook/blob/c58ff0b1e6d4fbbb0288605986ea57f18708e3ff/src/rest/mod.rs#L69) or [webhook multipart handler](https://github.com/tonkku107/slashook/blob/c58ff0b1e6d4fbbb0288605986ea57f18708e3ff/src/webhook/multipart.rs#L32).   
These functions (and fields) should also be added for the `File` struct as you'd only be able to potentially use these `Attachment` ones for editing (need to test if discord allows editing these fields).

**Review by tonkku107 — CHANGES_REQUESTED** [review]

**Review by tonkku107 — CHANGES_REQUESTED** [review]
There should be a function in `MessageResponse` to set the voice message flag

**Review by tonkku107 — APPROVED** [review]
LGTM

**Pull request merged**